os/binfmt/binfmt_exec.c

```diff
@@ -211,10 +211,9 @@ int exec(FAR const char *filename, FAR char *const *argv, FAR const struct symta
 	tcb->ram_start = (uint32_t)start_addr;
 
 	/* Initialize the MPU registers in tcb with suitable protection values */
-#ifdef CONFIG_ARMV7M_MPU
+#if defined(CONFIG_ARMV7M_MPU) && defined(CONFIG_ARCH_CHIP_IMXRT)
```

Comment:
i think chip specific code should not be present here.

Reply:
I agree @yashwanth686007 's opinion.
If this is not chip specific, you had better turn on the config of disabling binfmt in ARTIK.

I already asked @kishore-sn that why binfmt is enabled by default even though app_separation is not a default.
If we change `CONFIG_BINFMT_DISABLE` to `CONFIG_BINFTM_ENABLE`, there is no problem. I think.